apm-agent-core/src/main/java/co/elastic/apm/agent/bci/classloading/ExternalPluginClassLoader.java

```diff
@@ -53,6 +56,15 @@ public ExternalPluginClassLoader(File pluginJar, ClassLoader parent) throws IOEx
         }
     }
 
+    private static ClassLoader createFilteringClassLoader(ClassLoader parent) {
+        // The agent classloader contains "hidden" Otel dependencies due to the embedded metric SDK
+        // We force plugins to provide their own Otel version
+        return new DiscriminatingMultiParentClassLoader(
+            parent, not(startsWith("io.opentelemetry.")),
+            parent, not(startsWith("io.opentelemetry."))
```

Comment:
> 1. allow the second class loader to be `null`, in which case it will be ignored and the `DiscriminatingMultiParentClassLoader#discriminators` list will have a length of 1

For simplicity I'll go with this approach, but due to `null` usually referring to the bootstrap loader i'll add a different constructor which just takes a single parent as argument.